js/compiler_tests.ts

```diff
@@ -0,0 +1,466 @@
+    // be easily unwound.  So we will do what we can with the inner one and
+    // then just clear it...
+    mockDeps = undefined;
+    mockFactory = undefined;
+  };
+
+  const result = compilerInstance.run("foo/baz.ts", "/root/project");
+  assert(result instanceof ModuleMetaData);
+  // we have mocked that foo/bar.ts is already cached, so two fetches,
+  // but only a single cache
+  assertEqual(codeFetchStack.length, codeFetchLength + 2);
+  assertEqual(codeCacheStack.length, codeCacheLength + 1);
+  // because of the challenges with the way the module factories are generated
+  // we only get one invocation of the `globalEval` mock.
+  assertEqual(globalEvalStack.length, globalEvalLength + 1);
+  assert(factoryRun);
+});
+
+// TypeScript LanguageServiceHost APIs
+
+test(function compilerGetCompilationSettings() {
+  const result = compilerInstance.getCompilationSettings();
+  for (const key of [
+    "allowJs",
+    "module",
+    "outDir",
+    "inlineSourceMap",
+    "inlineSources",
+    "stripComments",
+    "target"
+  ]) {
+    assert(key in result, `Expected "${key}" in compiler options.`);
+  }
+  assertEqual(Object.keys(result).length, 7, "Expected only 7 keys.");
+});
+
+test(function compilerGetNewLine() {
+  const result = compilerInstance.getNewLine();
+  assertEqual(result, "\n", "Expected newline value of '\\n'.");
+});
+
+test(function compilerGetScriptFileNames() {
+  compilerInstance.run("foo/bar.ts", "/root/project");
+  const result = compilerInstance.getScriptFileNames();
+  assertEqual(result.length, 1, "Expected only a single filename.");
+  assertEqual(result[0], "/root/project/foo/bar.ts");
+});
+
+test(function compilerGetScriptKind() {
+  assertEqual(compilerInstance.getScriptKind("foo.ts"), ScriptKind.TS);
+  assertEqual(compilerInstance.getScriptKind("foo.d.ts"), ScriptKind.TS);
+  assertEqual(compilerInstance.getScriptKind("foo.js"), ScriptKind.JS);
+  assertEqual(compilerInstance.getScriptKind("foo.json"), ScriptKind.JSON);
+  assertEqual(compilerInstance.getScriptKind("foo.txt"), ScriptKind.JS);
+});
+
+test(function compilerGetScriptVersion() {
+  const moduleMetaData = compilerInstance.resolveModule(
+    "foo/bar.ts",
+    "/root/project"
+  );
+  assertEqual(
+    compilerInstance.getScriptVersion(moduleMetaData.fileName),
+    "1",
+    "Expected known module to have script version of 1"
+  );
+});
+
+test(function compilerGetScriptVersionUnknown() {
+  assertEqual(
+    compilerInstance.getScriptVersion("/root/project/unknown_module.ts"),
+    "",
+    "Expected unknown module to have an empty script version"
+  );
+});
+
+test(function compilerGetScriptSnapshot() {
+  const moduleMetaData = compilerInstance.resolveModule(
+    "foo/bar.ts",
+    "/root/project"
+  );
+  const result = compilerInstance.getScriptSnapshot(moduleMetaData.fileName);
+  assert(result != null, "Expected snapshot to be defined.");
+  assertEqual(result.getLength(), 87, "Expected a length of 87.");
```

Comment:
Instead of using 87, use fooBarTsSource.length. No need for excessively verbose assert messages either.

Reply:
Done